amr-wind/actuator_line/NREL5MW_ALM_BD/setup/README.md

```diff
@@ -0,0 +1,277 @@
+
+```bash
+$ AMRWINDFEDIR/utilities/downloadOFmodel.py nrel5mw_v402_rosco.yaml
+```
+
+The `downloadOFmodel.py` will pull the necessary files straight from the OpenFAST repo, ensure files are from the right OpenFAST version, and make edits to the OpenFAST parameters so that the turbine is correctly configured for this case.
+
+Then change the name of the directory to match what is used in the input file
+```bash
+$ mv NREL5MW_v402_ROSCO/ T0_NREL5MW_v402_ROSCO/
+```
+
+## Step 3: edit the file locations
+
+There are two edits required to the input file `NREL5MW_ALM_BD_OFv402.inp` in order to properly refer to the right locations.  The first is the location of the restart file: 
+
+```
+io.restart_file                          = /gpfs/lcheung/HFM/exawind-benchmarks/Neutral_ABL/chk30000
+```
+
+The second is the location of the boundary I/O files: 
+
+```
+ABL.bndry_file                           = /gpfs/lcheung/HFM/exawind-benchmarks/convective_abl/bndry_file
+```
+
+In each of these lines, change the path to match the location where the precursor directory was run.
+
+## Step 4: submit the run
+
+On many HPC platforms, a submission script needs to used to launch the case.  For slurm based systems, a submit script like this `submit.sh` file is used:
+```bash
+#!/bin/bash
+#SBATCH --nodes=8
+#SBATCH --time=47:59:59        # Wall clock time (HH:MM:SS) - once the job exceeds this time, the job will be terminated (default is 5 minutes)
+#SBATCH --job-name=NREL5MW # Name of job
+#SBATCH --partition=batch      # partition/queue name: short or batch
+#SBATCH --qos=normal           # Quality of Service: long, large, priority or normal 
+export nodes=$SLURM_JOB_NUM_NODES
+
+# Load any required modules here
+#module purge
+#module load ...
+
+export EXE=/projects/wind_uq/lcheung/AMRWindBuilds/hfm.20250211/amr-wind/build/amr_wind
+
+# Number MPI processes to run on each node (a.k.a. PPN)
+export cores=112
+export ncpus=$((nodes * cores))
+export OMP_PROC_BIND=spread 
+export OMP_PLACES=threads
+
+time mpiexec --bind-to core --npernode $cores --n $ncpus $EXE NREL5MW_ALM_BD_OFv402.inp 
+```
+
+Then, for slurm based queueing systems, submit the run with a command like:  
+```
+$ sbatch submit.sh
+```
+
+## Turbine and mesh settings
+
+In this section we provide some explanations and context for the turbine and mesh parameter settings used in the simulation.
+
+### Mesh and domain
+
+![](../results/images/NREL5MW_domain.png)
+
+The domain that is used in the NREL5MW case is the same as in the precursor ABL, namely one that has dimensions of 5120m x 5120m x 1920m.  The turbine base location is at (x,y) = (1800, 1800).  The background (level 0) grid uses a 10m resolution which is refined twice to reach 2.5m resolution at the rotor, for a total mesh size: 70.5M  Note that for the blade-resolved version of this run, two additional refinement zones are used, so the resolution at the rotor disk is 0.625m.
```

Comment:
Missing a period here after 70.5M.
Could also say "for a total mesh size of approximately 70.5M cells."